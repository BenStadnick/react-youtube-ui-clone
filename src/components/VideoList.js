import React from 'react';
import { Grid } from '@material-ui/core';
import VideoItem from './VideoItem';

const VideoList = ({ videos, onVideoSelect }) => {
  if(!videos.length) return <div>Unable to load recomendations</div>

  const listOfVideos = videos.map((video, id) => <VideoItem onVideoSelect={onVideoSelect} key={id} video={video} />)

  return (
    <Grid container spacing={10}>
      {listOfVideos}
    </Grid>
  )
}

export default VideoList;