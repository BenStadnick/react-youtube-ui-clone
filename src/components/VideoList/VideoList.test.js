import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import VideoList from './VideoList';

 // mock video api response
 const testTitle = 'test title';
 const testUrl = 'test url';
 const video = {
   snippet : {
     title : testTitle,
     thumbnails : {
       medium : {
         url : testUrl
       }
     }
   }
 };

describe('VideoList component', () => {
  it('did shallow render', () => {
    let wrapper = shallow(<VideoList />);
    expect(wrapper).toMatchSnapshot();

    let videos = [];
    wrapper = shallow(<VideoList videos={videos} />);
    expect(wrapper).toMatchSnapshot();


    videos = [video, video, video];
    wrapper = shallow(<VideoList videos={videos} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('displays the correct number of VideoItems based on \'video\' input', () => {
    let videos = [video, video, video];
    let wrapper = shallow(<VideoList videos={videos} />);
    expect(wrapper.children().length).toEqual(3);
  });
  
});