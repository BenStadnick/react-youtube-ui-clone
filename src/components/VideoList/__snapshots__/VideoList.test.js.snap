// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoList component did shallow render 1`] = `
<div>
  Unable to load recomendations
</div>
`;

exports[`VideoList component did shallow render 2`] = `
<div>
  Unable to load recomendations
</div>
`;

exports[`VideoList component did shallow render 3`] = `
<div>
  <VideoItem
    key="0"
    video={
      Object {
        "snippet": Object {
          "thumbnails": Object {
            "medium": Object {
              "url": "test url",
            },
          },
          "title": "test title",
        },
      }
    }
  />
  <VideoItem
    key="1"
    video={
      Object {
        "snippet": Object {
          "thumbnails": Object {
            "medium": Object {
              "url": "test url",
            },
          },
          "title": "test title",
        },
      }
    }
  />
  <VideoItem
    key="2"
    video={
      Object {
        "snippet": Object {
          "thumbnails": Object {
            "medium": Object {
              "url": "test url",
            },
          },
          "title": "test title",
        },
      }
    }
  />
</div>
`;
